packages/react-ai/src/components/AIConversation/views/Controls/InputControl.tsx

```diff
@@ -0,0 +1,157 @@
+    </svg>
+  ),
+  stroke: 'white',
+  'aria-hidden': true,
+  className: `${INPUT_BLOCK}__icon`,
+  width: '24',
+  height: '24',
+  viewBox: '0 0 16 16',
+  fill: 'none',
+  xmlns: 'http://www.w3.org/2000/svg',
+});
+
+const SendIcon = withBaseElementProps(Icon, sendIconProps);
+
+const SendButtonBase = withBaseElementProps(Button, {
+  'aria-label': 'Send message',
+  className: `${INPUT_BLOCK}__button ${INPUT_BLOCK}__button--send`,
+  disabled: false,
+});
+
+const SendButton: typeof SendButtonBase = React.forwardRef(
+  function SendButton(props, ref) {
+    // TODO should come from context
+    const isWaitingForResponse = false;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- TODO send message on click
+    const { input } = React.useContext(InputContext);
+    return (
+      <SendButtonBase
+        onClick={() => {}}
+        disabled={isWaitingForResponse}
+        {...props}
+        ref={ref}
+      />
+    );
+  }
+);
+
+const TextInputBase = withBaseElementProps(TextArea, {
+  className: `${INPUT_BLOCK}__input`,
+  placeholder: 'Message Raven',
+  id: `${INPUT_BLOCK}-text-input`,
+});
+
+const TextInput: typeof TextInputBase = React.forwardRef(
+  function TextInput(props, ref) {
+    const { input, setInput } = React.useContext(InputContext);
+    // TODO should come from context or prop
+    const isFirstMessage = false;
+
+    const debouncedSetInput = debounce((value: string) => {
+      setInput!(value);
+    }, TYPEAHEAD_DELAY_MS);
+
+    const onType = React.useCallback(
+      (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+        debouncedSetInput(event.target.value);
+      },
+      [debouncedSetInput] // Only re-create when debouncedSetInput changes
+    );
+
+    React.useEffect(() => {
+      const textarea = document.getElementById(`${INPUT_BLOCK}-text-input`);
+
+      const handleResize = () => {
+        if (textarea) {
+          textarea.style.height = 'auto';
+          textarea.style.height = `${textarea.scrollHeight}px`;
+        }
+      };
+
+      if (textarea && textarea instanceof HTMLTextAreaElement) {
+        textarea.addEventListener('input', handleResize);
+        handleResize();
+      }
+
+      return () => {
+        if (textarea) {
+          textarea.removeEventListener('input', handleResize);
+        }
+      };
+    }, [input]);
+
+    return (
+      <TextInputBase
+        data-testid="text-input"
+        onChange={(e) => setInput && onType(e)}
+        {...(isFirstMessage ? { placeholder: 'Ask anything...' } : {})}
+        {...props}
+        ref={ref}
+      />
+    );
+  }
+);
+
+const Container = withBaseElementProps(View, {
+  className: `${INPUT_BLOCK}__container`,
+});
+
+export const InputControl: InputControl = () => {
+  return (
+    <Container>
+      <AttachFileControl />
+      <TextInput />
+      <SendButton>
+        <SendIcon />
+      </SendButton>
+    </Container>
+  );
+};
+
+InputControl.AttachFile = AttachFileControl;
```

Comment:
a hidden input needed for core functionality is not composable and needs to live inside and be scoped to the component that actually needs it